convert_deployment_to_csv.py

```diff
@@ -0,0 +1,172 @@
+import sys
+from os import listdir
+from os.path import isfile, join, splitext
+import yaml
+import json
+
+
+SCHEMAS_DIRECTORY = "schemas"
+TABLE_MARGIN_TOP = 1
+TABLE_MARGIN_LEFT = 1
+
+
+def usage():
+    print "Converts a deployment configuration folder into a CSV file (spreadsheet)"
+    print ""
+    print "Usage:"
+    print "    " + " ".join([sys.argv[0],
+                             "<deployment_folder>",
+                             "<csv_file>"])
+    print ""
+
+
+def get_all_schemas(deployment_folder):
+    schemas = [splitext(f)[0] for f in listdir(deployment_folder)
+               if isfile(join(deployment_folder, f))]
+    return schemas
+
+
+def getValuesFromDeployment(deployment_folder, deployment):
+    return yaml.safe_load(open(join(deployment_folder, deployment + '.yml')))
+
+
+def read_schema(schema_name):
+    file_name = schema_name + ".json"
+    file_path = join(SCHEMAS_DIRECTORY, file_name)
+    with open(file_path, "r") as f:
+        schema_str = f.read().decode("utf-8")
+    try:
+        return json.loads(schema_str)
+    except Exception as e:
+        raise Exception("Could not parse schema: %s\n%s" % (
+            file_name, str(e)))
+
+
+def write_csv_file(filename, content):
+    f = open(filename, "w")
+    f.write(content)
+
+
+def escape_line(line):
+    if '"' in line or ',' in line:
+        return '"' + line.replace('"', '""') + '"'
+    return line
+
+
+def add_content(content):
+    csvContent = ''
+    for item in content:
+        if type(item) == dict:
+            csvContent += (write_table(item).encode("utf-8")) + '\n'
+        else:
+            csvContent += escape_line(item) + "," + '\n'
+    return csvContent
+
+
+def write_fields(lines, schema, table):
+    data = table['data']
+
+    if schema["type"] == "array":
+        schema_props = schema["items"]["properties"]
+        required = schema["items"].get("required")
+    else:
+        schema_props = schema["properties"]
+        required = schema.get("required")
+
+    dataDict = {}
+    dataDictList = []
+    if type(data) == list:
+        dataDictList = data
+    else:
+        dataDict = data
+    fields = required
+    for k in dataDict.keys():
```

Comment:
Using the key values of a dict has a random order.  Can you refactor this so that the "propertyOrder" field defines the order of the fields in the example.  This is how I did it in other libraries:

`        for name, field in sorted(iter(properties.items()),
                                  key=lambda v: v[1]["propertyOrder"]):
`